packages/fxa-auth-server/lib/payments/paypal.ts

```diff
@@ -397,10 +397,10 @@ export class PayPalHelper {
    * @param invoice
    */
   public processZeroInvoice(invoice: Stripe.Invoice) {
-    return Promise.all([
-      this.stripeHelper.finalizeInvoice(invoice),
-      this.stripeHelper.payInvoiceOutOfBand(invoice),
-    ]);
+    // It appears for subscriptions that do not require payment, the invoice
+    // transitions to paid automatially.
+    // https://stripe.com/docs/billing/invoices/subscription#sub-invoice-lifecycle
+    return this.stripeHelper.finalizeInvoice(invoice);
```

Comment:
Before removing `payInvoiceOutOfBand` it would actually error on that line, with a message along the lines of "This has already been paid".

That's why I thought it's safe to remove, but I'm not sure if it applies in all scenarios with zero invoices.

Reply:
Ah I see.  I think it would be nice to still mark it as paid out of band (not a blocker though).

Reply:
Unfortunately even with `auto_advance` set to false on the finalize call, it looks like the invoice is still returned as `paid` after finalizes completes.